packages/legend-application/src/components/shared/PackageableElementOptionRenderer.tsx

```diff
@@ -14,23 +14,46 @@
  * limitations under the License.
  */
 
-import type { PackageableElement } from '@finos/legend-graph';
+import type {
+  PackageableElement,
+  GraphManagerState,
+} from '@finos/legend-graph';
 import type { PackageableElementOption } from '../../stores/shared/PackageableElementOption.js';
 
-export const getPackageableElementOptionalFormatter = (props?: {
+export const getPackageableElementOptionalFormatter = (props: {
   darkMode?: boolean;
+  graphManagerStatePackage: GraphManagerState;
 }): ((
   option: PackageableElementOption<PackageableElement>,
 ) => React.ReactNode) =>
   function PackageableElementOptionLabel(
     option: PackageableElementOption<PackageableElement>,
   ): React.ReactNode {
-    const className = props?.darkMode
+    const className = props.darkMode
       ? 'packageable-element-format-option-label--dark'
       : 'packageable-element-format-option-label';
 
+    const element = option.value;
+
+    const optionColor = props.graphManagerStatePackage.isPrimitiveTypeElement(
+      element,
+    )
+      ? 'primitive'
+      : props.graphManagerStatePackage.isSystemElement(element)
+      ? 'system'
+      : props.graphManagerStatePackage.isGeneratedElement(element)
+      ? 'generated'
+      : props.graphManagerStatePackage.isMainElement(element)
+      ? 'generated'
+      : props.graphManagerStatePackage.isDependencyElement(element)
+      ? 'dependency'
+      : '';
+
     return (
       <div className={className}>
+        <span
```

Comment:
Let's use `div` instead of `span`, I think we're styling this using `flex` anyway, `span` doesn't have much value here 

Also, let's add more info upon hover here to say this element belongs to `system`, etc.